Currently build fails with

```
[ 91%] Building CXX object CMakeFiles/hyprpolkitagent.dir/build/.qt/rcc/qrc_hyprpolkitagent_raw_qml_0.cpp.o
/build/source/src/core/Agent.cpp:4:10: fatal error: print: No such file or directory
    4 | #include <print>
      |          ^~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/hyprpolkitagent.dir/build.make:180: CMakeFiles/hyprpolkitagent.dir/src/core/Agent.cpp.o] Error 1
```

--- a/pkgs/by-name/hy/hyprpolkitagent/package.nix
+++ b/pkgs/by-name/hy/hyprpolkitagent/package.nix
@@ -1,0 +1,50 @@
+{
+  lib,
+  stdenv,
+  cmake,
+  pkg-config,
+  fetchFromGitHub,
+  hyprutils,
+  kdePackages,
+  polkit,
+  qt6,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "hyprpolkitagent";
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "hyprwm";
+    repo = "hyprpolkitagent";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-fhG6YqByDW0DAixXXX6AwTJOH3MqDlQ2XrVvpusZ3Ek=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    qt6.wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    hyprutils
+    kdePackages.kirigami-addons
+    kdePackages.polkit-qt-1
+    polkit
+    qt6.qtbase
+    qt6.qtsvg
+    qt6.qtwayland
+  ];
+
+  meta = {
+    description = "Polkit authentication agent written in QT/QML";
+    homepage = "https://github.com/hyprwm/hyprpolkitagent";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [
+      fufexan
+      johnrtitor
+    ];
+    mainProgram = "hyprpolkitagent";
+    platforms = lib.platforms.linux;
+  };
+})

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -5218,6 +5218,10 @@ with pkgs;
 
   hyprpicker = callPackage ../applications/window-managers/hyprwm/hyprpicker { };
 
+  hyprpolkitagent = callPackage ../by-name/hy/hyprpolkitagent/package.nix {
+    stdenv = gcc14Stdenv;
+  };
+
   hyprshade = python311Packages.callPackage ../applications/window-managers/hyprwm/hyprshade { };
 
   hyprlandPlugins = recurseIntoAttrs (callPackage ../applications/window-managers/hyprwm/hyprland-plugins { });

